src/tss2-sys/sysapi_util.c

```diff
@@ -304,3 +303,111 @@ int GetNumResponseHandles(TPM2_CC commandCode)
 {
     return GetNumHandles(commandCode, 0);
 }
+
+#ifdef DISABLE_WEAK_CRYPTO
+bool IsAlgorithWeak(TPM2_ALG_ID algorithm, TPM2_KEY_SIZE key_size)
+{
+    switch (algorithm) {
+        case TPM2_ALG_RSA:
+            if (key_size <= 2048)
+                return true;
+        break;
+        case TPM2_ALG_AES:
+            if (key_size <= 128)
```

Comment:
Why is this considered weak? The smallest possible key size for AES is 128 bits, and that is generally still considered secure, isn't it? There are no known attacks that significantly reduce the strength of AES (certainly not below the NIST limit of 112 bits).

Reply:
Yes, symmetric key of 128 provides effective security strength of 128 bits and is safe to use.
This should only disable anything that is 112 and smaller. This will make most of the ciphers (AES, SM4 and Camellia) usable with all supported key sizes (128, 192, and 256), so this check will probably be redundant, but the the spec also defines a generic TPM2_ALG_SYMCIPHER ALG_ID and the algorithm registry [1] also calls out Triple DES I have updated the check to cover all ciphers and bail out on key < 128. Thanks for pointing this out.
[1] https://trustedcomputinggroup.org/wp-content/uploads/TCG-_Algorithm_Registry_Rev_1.27_FinalPublication.pdf